src/iop/scalepixels.c

```diff
@@ -147,16 +143,26 @@ void modify_roi_in(dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const d
 {
   *roi_in = *roi_out;
 
-  float xy[2] = { roi_in->x, roi_in->y };
-  float wh[2] = { roi_in->width, roi_in->height };
+  // If possible try to get an image that's strictly larger than what we want to output
+  float hw[2] = {roi_out->height, roi_out->width};
+  transform(piece, hw); // transform() is used reversed here intentionally
+  roi_in->height = hw[0];
+  roi_in->width = hw[1];
 
-  backtransform(piece, xy);
-  backtransform(piece, wh);
+  float reduction_ratio = MAX(hw[0] / (piece->iheight * 1.0f), hw[1] / (piece->iwidth * 1.0f));
```

Comment:
Ah, I see what you mean. It doesn't seem to make a difference, even with low_quality_thumbs. I've pushed the change now.